src/app/api/auth/[...nextauth]/route.ts

```diff
@@ -0,0 +1,43 @@
+import NextAuth, { AuthOptions } from 'next-auth';
```

Comment:
I believe that this makes it more clear. I have both an auth and client.ts file in the api folder, but I think that it's not necessary.  It is a common approach, though.

Reply:
sure sure. The additional API layer (this is what you did) is great when you want to separate some transformations to the "backend", cause this will run only on the server. It is also great when you have some API keys that you don't want to expose to the client. I think we do neither of those.

On the other hand, the API "folder", so basically a random folder called "api" or whatever else is just a separation of the api calls from the components. So you could reuse the same, named, endpoints. Nothing fancy, something like this:

![CleanShot 2023-08-14 at 12 48 09](https://github.com/GLObus303/sportujspolu/assets/12488648/1ba9a73d-7705-4527-9d73-96e87fcd260d)


I think you can in the end use both. if you layer it correctly, changing from `ky` to `axios` or anything else can be a matter of changing import and single base function

![CleanShot 2023-08-14 at 12 49 21](https://github.com/GLObus303/sportujspolu/assets/12488648/177cf146-682b-4b82-9ea5-fb3bf2740fff)

unlike having 1 million places where you compose the api calls in components over and over

Reply:
I see, I will try to reconsider and rearrange that when rewriting to raw auth. 🤓